packages/data-table-server/src/app/createApp.ts

```diff
@@ -22,6 +29,11 @@ export function createApp(database = new TupaiaDatabase()) {
       attachDataTableToContext,
       handleWith(FetchDataRoute),
     )
+    .post<FetchPreviewDataRequest>(
+      'dataTable/fetchPreviewData',
+      attachDataTableFromPreviewToContext,
```

Comment:
Why is this a middleware? Seems like it's just used once in this route. The other middleware is used multiple times.

Reply:
I am thinking to keep the logic consistent to the other middleware. Does it look confusing?